types.ts

```diff
@@ -0,0 +1,11 @@
+import { createElement } from 'react'
+
+export interface Scope {}
+
+export type Components = Parameters<typeof createElement>[1]
```

Comment:
This is some crazy wizardry here ✨ - could you help me to understand what's going on? afaik, the `Parameters` utility creates a type out of the arguments to a function, but you're passing in the results of a `typeof` operation which I'd think would return `"function"`?

Reply:
@jescalan No, the `typeof` here means the types of the parameters.

For example:

```ts
declare function f1(arg: { a: number; b: string }): void;

type T3 = Parameters<typeof f1>;
//    ^ = type T3 = [arg: {
    a: number;
    b: string;
}]
```

Reply:
@jescalan No, the `typeof` here means the types of the parameters.

For example:

```ts
declare function f1(arg: { a: number; b: string }): void;

type T3 = Parameters<typeof f1>;
//    ^ = type T3 = [arg: {
    a: number;
    b: string;
}]
```